### DISCUSSION TITLE
Q&A General

### DISCUSSION THREAD

**Discussion by pavlobu** [pavlobu/deskreen#89] [Q&A]
Hi all. 👋 This thread is for general questions about Deskreen. You are also welcome to answer unanswered questions here.

**Comment by DCrawley94** [comment]
Hiya, I've been looking into options for using my laptop as a "second monitor" for my desktop PC. I know that this is quite an awkward thing to do on Linux.

I mainly just need it for screen-sharing from my laptop as I don't have a webcam for my desktop computer. If I can do that then I don't need to access the necessary files remotely.

**Reply by pavlobu** [reply]
Sure you can do this with Deskreen. Open web browser on your laptop, type IP adress that is displayed under qr code and you are good to go :)

**Reply by DCrawley94** [reply]
Thanks for the quick reply! I'll give this a go :)

**Comment by yuzhichang** [comment]
Does it behave like interactive remote-desktop?

**Reply by pavlobu** [reply]
Hi, no not at the current version. Will be added in sone future release, but not soon

**Comment by Skaifer** [comment]
Hi, awesome app! Ability to use laptop as second monitor wirelessly blows my mind, I need that in my life🤯. Have a question, do I need to buy one HDMI Dummy for client, or two for both sides?

**Reply by pavlobu** [reply]
You need Only one for your computer where app Deskreen runs

**Reply by jcc10** [reply]
Actually, you can use something like [itsmikethetech/Virtual-Display-Driver](https://github.com/itsmikethetech/Virtual-Display-Driver?tab=readme-ov-file) to do it in software. I'm doing it right now actually.

**Comment by nicolaiandler** [comment]
Hi Paul,

The local host 127.0.0.1 simulation worked on my macbook but my iPad is not responding - the URL in safari on iPad does not respond… times out/served stopped responding

So I am therefore stuck with the wifi router firewall problem that you describe as the only other reason. There is no anti-virus on my mac. - however I dont know what settings of the firewall to change for deskreen to work…for my ipad to connect  - obviously not disabling the complete firewall for deskreen.

I have a d-Link Dir 825 router… 
http://www.dlink.cc/d-link-router/d-link-dir-825-firewall-settings-and-routing-configuration.html
http://screenshots.portforward.com/routers/Dlink/DIR-825/Firewall_Settings.htm
http://screenshots.portforward.com/routers/Dlink/DIR-825/

can you give me a prompt/brief instruction 

I want to use powerpoint from my macbook > extended to ipad then flip the presentation to display it in a telepromter !

My system
Mac OS 10.14.6
iPad 14.3
no intention of using sidecar !

I do not have a virtual display adapter installed - my understanding is that it is not really required ??  

truly appreciate your time and dev work….happy to do a donation if I can get deskreen actually to work for me !!

Best regards
Nicolai

**Reply by soaringleefighting** [reply]
I encountered this question, too. 
my system: Mac OS 11.3  iPad 10.3.3
When I open web browser on my tablet, type IP adress that is displayed under qr code, the tablet is not responding, with no prompt.

**Comment by nicolaiandler** [comment]
my issue was the router wifi frequency 

it only works for me on a 2.4 but not the 5 ghz frequency wifi

**Reply by soaringleefighting** [reply]
Thanks for the quick reply! I will try it. I doubt that my another iPad mini (ios 14.0) works well, but iPad Air 2(ios 10.3.3) not work.

**Reply by soaringleefighting** [reply]
It seems that iOS 10.3.3 not work.

**Reply by ZheFrench** [reply]
Same story with Ipad 2(ios 14.6) from ubuntu 16.

**Comment by PS-Professional** [comment]
Hello
I am very interested in using Deskreen for my daily works but while I'm using my mobile hotspot (in most cases), Deskreen says: "No WiFi and LAN connection".
Is it possible to use Deskreen in this kind of network connection?

**Reply by pavlobu** [reply]
Yes its possible, you need to have both of your devices connected to this mobile network. Laptop and display device should be in same network. If you are trying to use your mobile for hotspot and as a viewing display, you need to enter your laptop ip on your bomile browser.

http://(your.laptop.ip.number):3131/(12345)

Replace values between () with your numbers

**Reply by JonnyOnlineYT** [reply]
> Yes its possible, you need to have both of your devices connected to this mobile network. Laptop and display device should be in same network. If you are trying to use your mobile for hotspot and as a viewing display, you need to enter your laptop ip on your bomile browser.
> 
> http://(your.laptop.ip.number):3131/(12345)
> 
> Replace values between () with your numbers

How should I get these numbers (in your example 12345) tho?
The window is getting blocked by that „No WiFi and LAN connection“ screen.

**Comment by Fancy2209** [comment]
Can I use the touch screen on device I am using as a 2nd screen  (Tablet) trough Deskreen?

**Reply by pavlobu** [reply]
Unfortunately not

**Comment by samggreenberg** [comment]
Hi. I'm trying to run Deskreen FROM a Chromebook (instead of from Windows or MacOS) and encountering some errors. :(

My virtual display adapter is working and I can install the deskreen.deb, but when running deskreen, I get:
```
samgreenberg@penguin:~$ deskreen
08:39:05.039 › /opt/Deskreen/resources/app.asar/main.prod.js : Deskreen signaling server is online at port 3131
08:39:05.462 › APPIMAGE env is not defined, current application is not an AppImage
libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)
MESA-LOADER: failed to retrieve device information
MESA-LOADER: failed to retrieve device information
MESA-LOADER: failed to retrieve device information
[1983:0625/083905.966571:ERROR:gbm_wrapper.cc(275)] Failed to export buffer to dma_buf: No such file or directory (2)
```
That last line then iterates through a bunch of those numbers I don't understand with the same "Failed to export buffer" error.

Any tips for how to run Deskreen on a Chromebook?

**Reply by pavlobu** [reply]
Try to download and use .AppImage file instead of .deb

**Reply by samggreenberg** [reply]
When I use the .AppImage as root, I get:
```
samgreenberg@penguin:~$ sudo ./Deskreen-2.0.3.AppImage --no-sandbox
12:00:25.271 › /tmp/.mount_DeskresbwoDh/resources/app.asar/main.prod.js : Deskreen signaling server is online at port 3131
No protocol specified
[570:0625/120025.330718:ERROR:ozone_platform_x11.cc(248)] Missing X server or $DISPLAY
[570:0625/120025.331181:ERROR:env.cc(225)] The platform failed to initialize.  Exiting.
The futex facility returned an unexpected error code.
Aborted
```
When I use the .AppImage without root, I get:
```
12:06:24.753 › /tmp/.mount_DeskreWi4Run/resources/app.asar/main.prod.js : Deskreen signaling server is online at port 3131
libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)
12:06:26.439 › Checking for update
MESA-LOADER: failed to retrieve device information
MESA-LOADER: failed to retrieve device information
MESA-LOADER: failed to retrieve device information
[1090:0625/120626.593726:ERROR:gbm_wrapper.cc(275)] Failed to export buffer to dma_buf: No such file or directory (2)
...
[1090:0625/120626.623753:ERROR:gbm_wrapper.cc(275)] Failed to export buffer to dma_buf: No such file or directory (2)
12:06:26.629 › Generated new staging user ID: f2ef0949-398d-51eb-8160-1232a08af885
12:06:27.578 › Update for version 2.0.3 is not available (latest version: 2.0.3, downgrade is disallowed).
12:06:27.584 › checkForUpdatesAndNotify called, downloadPromise is null
[1090:0625/120628.810958:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[1090:0625/120629.229947:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
[1090:0625/120632.580493:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times!
```


P.S. Thanks for your help! I really appreciate it. I donated at the savelife link.

**Reply by pavlobu** [reply]
Hmm. Try to start it not from a terminal but with a double click from desktop environnement and see what happens

**Reply by pavlobu** [reply]
Thanks for support btw!

**Reply by jamesshannon** [reply]
Hi @pavlobu Just got a chromebook so I'm trying to solve this myself. I'm getting the same error as Sam. Not sure if he got it working? @pavlobu -- Not sure how much you know about ChromeOS, but the linux apps are run inside of VMs, which imposes some unexpected [runtime "features"](https://chromeos.dev/en/linux/linux-on-chromeos-deep-dive#runtime-features). I think the `dma_buf` error messages are related to GPU access, which would sort of make sense. (Though I've also [enabled GPU acceleratio in the linux VM](https://chromeready.com/12488/enable-gpu-acceleration-chromebooks-linux/). Given that you're the expert in how deskreen works would you expect it to work within the ChromeOS Linux runtime at all? Anything I can do to help you debug?

**Reply by MrDetox** [reply]
@jamesshannon I am having the same issue, did you ever manage to figure it out?

**Comment by Spirarementum** [comment]
Hi,
Is there a deb package for Ubuntu or a way to compile it from github to install deskreen ?
Thanks

**Reply by PS-Professional** [reply]
Yes, there is deb package available. Just check releases and download it.

**Reply by Spirarementum** [reply]
Ah...got it now !
Thanks

**Comment by EDIflyer** [comment]
What does the 'default video player' option do in the browser? I'm running Chrome for Android and haven't noticed any different if I have it on or off! 

I've also noticed although flip video works the scaling options are all greyed out, wasn't sure why that was?

Thanks again for a great app 😃

**Comment by poorhackerman** [comment]
Hellooo! 

Quick question - Is it possible to toggle fullscreen mode in remote desktop device without turning it on manually? The problem is that my desktop device has no keyboard and mouse port (I can only operate browser via terminal). This will be very handfull to me! Thanks in advice!

**Comment by shodanx2** [comment]
What kind of latency does deskscreen get, say a windows server and a raspberry pi client ? 
Will it stay below 50ms on wired ethernet ?

**Comment by cogotemartinez** [comment]
Hi, i need help with an issue i get when im trying to connect a windows 10 pc and a linux. If i want to show my win10 screen (use it as server) it works fine (linux connects and i can see the screen); but vice versa it doesnt work (when showing linux screen), win10 doesnt connect
![2023-01-09 18_05_55-Greenshot](https://user-images.githubusercontent.com/62314924/211408329-ab3ec70a-135c-4560-9ff8-df0fbfb88af4.png)

**Comment by pavlobu** [comment]
@tommar21
it may be a VPN problem. unfortunately when it's active it can break things

**Reply by cogotemartinez** [reply]
But i dont have any vpn, what can i do?

**Reply by cogotemartinez** [reply]
I cant connect to linux from any device, so the problem is sth in my linux pc, help me pls

**Comment by bxueaa** [comment]
Hi! Thank you very much for making this great software.
I wonder if I can make address a fixed suffix instead of randomly assigning a number every time

**Reply by jazzcrack** [reply]
Yes, I need a fix port number too. Would be cool to save an connection with all options (kind of screen/app windows and port) for later use.

**Reply by GabSabbath** [reply]
Is this implemented yet? Is there a way to fix the entire URL so that if Deskreen is running, I can quicky connect my second device without retyping the last numbers?

**Comment by Muggleborn** [comment]
Hi!  Rather than using a router to create a LAN, could I use internet sharing from my Linux box?  And if so, could I use a USB Ethernet adapter to a Chromebook to make a wired LAN directly to the Host?

**Comment by BarleyXu** [comment]
Hi, great job!!!
Could you please explain how Deskreen implements the functionality of extending the screen (dragging windows)? What is the difference between implementing this and screen mirroring? Why is it necessary to insert a virtual display adapter to achieve the extended screen functionality?

**Comment by limavicente** [comment]
Hi Pavlo. I am very satisfied with the software.
I use Linux and did this from the command line:
xrandr --addmode VIRTUAL1 1360x768
So, my S.O. identified a new screen. So I indicated this new screen for the deskscreen to send to a TV.
Now I have 3 extended monitors [LG Tv (hdmi), Laptop screen, VIRTUAL1]

Vicente Lima

**Comment by jackyhkust** [comment]
Hi Pavlo, it is a really great one. Thanks. I have a short question. Is there any option that we can force the deskreen client going to fullscreen in the address (and also set the Default Video player to True)? Now I can ssh into my client machine and and visit the address in a single script. Still, i have to switch to my client machine for the fullscreen. It will be nice if I can skip that step.

**Comment by vanshchoubey** [comment]
Hi Pavlo , i am having a problem with deskreen . my app is not opening it showing " no wifi or LAN connection " . I gone throught with many of the documentation and even tried turnoff firewall and alot .Could you solve my problem

**Reply by Vindobonus** [reply]
> Hi Pavlo , i am having a problem with deskreen . my app is not opening it showing " no wifi or LAN connection " . I gone throught with many of the documentation and even tried turnoff firewall and alot .Could you solve my problem

For me it is only running with Win10, when I try it on Win11 on the same PC in the same network there is only the "no wifior LAN connection". I disabled the Firewall no VPN - still not working. Has anybody used it with Win11?

**Reply by Doollitle** [reply]
I have a multiboot one partition with a upgraded windows 10 to windows 11 and a partition with a clean installed windows 11. 
On the clean installed i have the same issue, "No WiFi and LAN connection" message and can't do anything. 
On the upgraded to windows 11 it seems to work fine.

- Disabling Firewall same issue
- Uninstalled ProtonVPN same issue
- static IP address same issue

So still the same issue, if anyone finds out why, please help.

**Comment by Liftyee** [comment]
I have a 16:10 laptop but am projecting to a 16:9 low res screen, I would like to make the most of the pixels and crop out the taskbar. Is there a way to make Deskreen crop out part of the input before streaming?
